src/UmbracoExamine.PDF/PdfIndexValueSetBuilder.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Examine;
+using System.Collections.Generic;
+using Umbraco.Core;
+using Umbraco.Core.IO;
+using Umbraco.Core.Models;
+using Umbraco.Examine;
+
+namespace UmbracoExamine.PDF
+{
+    public interface IPdfIndexValueSetBuilder : IValueSetBuilder<IMedia> { }
+
+    /// <summary>
+    /// Builds a ValueSet for PDF Documents
+    /// </summary>
+    public class PdfIndexValueSetBuilder : IPdfIndexValueSetBuilder
+    {
+        private PdfTextService _pdfTextService;
+
+        public PdfIndexValueSetBuilder(PdfTextService pdfTextService)
+        {
+            _pdfTextService = pdfTextService;
+        }
+        public IEnumerable<ValueSet> GetValueSets(params IMedia[] content)
+        {
+            foreach (var item in content)
+            {
+                var umbracoFile = item.GetValue<string>(Constants.Conventions.Media.File);
+                if (string.IsNullOrWhiteSpace(umbracoFile)) continue;
+                var filepath = IOHelper.MapPath(umbracoFile);
+                var fileTextContent = ExtractTextFromFile(filepath);
+                var indexValues = new Dictionary<string, object>
+                {
+                    ["fileTextContent"] = fileTextContent,
+                    ["__IndexType"] = IndexTypes.Media,
+                    ["__NodeId"] = item.Id
+                };
+
+                var valueSet = new ValueSet(item.Id.ToString(), IndexTypes.Media, indexValues);
+
+                yield return valueSet;
+            }
+        }
+
+        private string ExtractTextFromFile(string filePath)
+        {
+            return _pdfTextService.ExtractText(filePath);
```

Comment:
Is it possible this could fail? do we need a try/catch?

Reply:
In theory everything could happen.. Depends on the implementation of `IPdfTextExtractor`

In the current example actually `NotImplementedException` could be called, so a try cache(exception) is probably needed